### PR TITLE
Scene: Add background blur with PMREM.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: #23712

**Description**

This is the alternative implementation of #23712 but via PMREMs.

### CODE REVIEW

File: src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export default /* glsl */`
 #ifdef USE_ENVMAP
 
+	uniform float envMapBlurriness;
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js:4]: We are not blurring the environment map. We are blurring the background.

**Comment by WestLangley** [comment]
I support your PR in principle, but not how it has been shoehorned into the existing shaders.

I have been trying to clean up the shaders, and this is not helping.

I am hopeful that if this is merged, we can work together on some shader refactoring.

File: src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js (hunk 1 above)
Comment (Mugen87) [reply src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js]: Updated!

**Comment by Mugen87** [comment]
Definitely. IMO when adding new features, there is no need to make the implementation perfect right from the beginning. We can use refactoring to fine-tune things.

File: src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export default /* glsl */`
 #ifdef USE_ENVMAP
 
+	uniform float backgroundBlurriness;
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js:4]: It should be clear now that this does not belong here.

The background needs its own shader chunks.

I hope we can agree that a coordinated effort at refactoring is in order.

File: src/renderers/shaders/ShaderChunk/envmap_fragment.glsl.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ export default /* glsl */`
 
 	#elif defined( ENVMAP_TYPE_CUBE_UV )
 
-		vec4 envColor = textureCubeUV( envMap, reflectVec, 0.0 );
+		vec4 envColor = textureCubeUV( envMap, reflectVec, backgroundBlurriness );
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/envmap_fragment.glsl.js:43]: This shader chunk is used only by Basic, Lambert, Phong, Toon, and Gouraud, none of which support ENVMAP_TYPE_CUBE_UV.

This shader chunk was designed for reflections, not for texturing a background box.

File: src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js (hunk 2 above)
Comment (WestLangley) [reply src/renderers/shaders/ShaderChunk/envmap_common_pars_fragment.glsl.js:4]: In a later, PR, of course.

File: src/renderers/shaders/ShaderChunk/envmap_fragment.glsl.js (hunk 3 above)
Comment (mrdoob) [reply src/renderers/shaders/ShaderChunk/envmap_fragment.glsl.js:43]: How do you think we should do it instead?

Comment (WestLangley) [reply src/renderers/shaders/ShaderChunk/envmap_fragment.glsl.js:43]: @mrdoob I think this `elif` should be removed from this shader chunk, and the background code should have its own chunks.

Let's merge the feature set we want first, then address these issues.

**Comment by Mugen87** [comment]
Updated builds: 4e93f9b3183bd28d888b3ff4d379a3e7b50b5e58

**Comment by WestLangley** [comment]
>ShaderLib.background is only used for simple textured backgrounds. Are you okay with extending it so it also handles env maps?

I think that would be OK. I am assuming it would be used exclusively by `scene.background`.

>so it also handles env maps...

Nit: I would prefer to say it this way: "so it also handles texture cubes and equirectangular textures".

**Pull request merged**